tests/test_sdk_client.py

```diff
@@ -54,99 +64,84 @@ def test_create_sdk_when_profile_has_ssl_errors_disabled_sets_py42_setting_and_p
+        create_sdk(mock_profile_with_password, False)
 
     assert "Invalid credentials for user" in err.value.message
     assert sdk_logger.log_error.call_count == 1
-    assert "Failure in HTTP call" in sdk_logger.log_error.call_args[0][0]
+    assert "Failure in HTTP call" in str(sdk_logger.log_error.call_args[0][0])
 
 
 def test_create_sdk_when_connection_exception_occurs_raises_and_logs_cli_error(
-    sdk_logger, mock_sdk_factory
+    sdk_logger, mock_sdk_factory, mock_profile_with_password
 ):
     mock_sdk_factory.side_effect = ConnectionError("connection message")
-    profile = create_mock_profile()
 
-    def mock_get_password():
-        return "Test Password"
-
-    profile.get_password = mock_get_password
     with pytest.raises(LoggedCLIError) as err:
-        create_sdk(profile, False)
+        create_sdk(mock_profile_with_password, False)
 
     assert "Problem connecting to" in err.value.message
     assert sdk_logger.log_error.call_count == 1
-    assert "connection message" in sdk_logger.log_error.call_args[0][0]
+    assert "connection message" in str(sdk_logger.log_error.call_args[0][0])
 
 
 def test_create_sdk_when_unknown_exception_occurs_raises_and_logs_cli_error(
-    sdk_logger, mock_sdk_factory
+    sdk_logger, mock_sdk_factory, mock_profile_with_password
 ):
     mock_sdk_factory.side_effect = Exception("test message")
-    profile = create_mock_profile()
-
-    def mock_get_password():
-        return "Test Password"
 
-    profile.get_password = mock_get_password
     with pytest.raises(LoggedCLIError) as err:
-        create_sdk(profile, False)
+        create_sdk(mock_profile_with_password, False)
 
     assert "Unknown problem validating" in err.value.message
     assert sdk_logger.log_error.call_count == 1
-    assert "test message" in sdk_logger.log_error.call_args[0][0]
-
+    assert "test message" in str(sdk_logger.log_error.call_args[0][0])
 
-def test_create_sdk_when_told_to_debug_turns_on_debug(mock_sdk_factory):
-    profile = create_mock_profile()
-
-    def mock_get_password():
-        return "Test Password"
 
-    profile.get_password = mock_get_password
-    create_sdk(profile, True)
+def test_create_sdk_when_told_to_debug_turns_on_debug(
+    mock_sdk_factory, mock_profile_with_password
+):
+    create_sdk(mock_profile_with_password, True)
     assert py42.settings.debug.level == debug.DEBUG
 
 
-def test_validate_connection_uses_given_credentials(mock_sdk_factory):
-    assert validate_connection("Authority", "Test", "Password", None)
-    mock_sdk_factory.assert_called_once_with("Authority", "Test", "Password", totp=None)
+def test_create_sdk_uses_given_credentials(
+    mock_sdk_factory, mock_profile_with_password
+):
+    create_sdk(mock_profile_with_password, False)
+    mock_sdk_factory.assert_called_once_with(
+        "example.com", "foo", "Test Password", totp=None
+    )
 
 
 def test_validate_connection_when_mfa_required_exception_raised_prompts_for_totp(
```

Comment:
Forgot to rename some of these `test_create_sdk_` rather than `test_validate_connection_`